invokeai/backend/model_manager/load/model_loaders/ip_adapter.py

```diff
@@ -24,13 +25,13 @@ class IPAdapterInvokeAILoader(ModelLoader):
 
     def _load_model(
         self,
-        model_path: Path,
-        model_variant: Optional[ModelRepoVariant] = None,
+        config: AnyModelConfig,
         submodel_type: Optional[SubModelType] = None,
     ) -> AnyModel:
         if submodel_type is not None:
             raise ValueError("There are no submodels in an IP-Adapter model.")
-        model = build_ip_adapter(
+        model_path = Path(config.path)
+        model: RawModel = build_ip_adapter(
```

Comment:
I think the annotation here of `: RawModel` is unnecessary

Reply:
Mypy complains of a type mismatch on the return statement unless it is there.

Reply:
Hmm, my editor is fine with it, but it uses pyright instead of mypy.

Is `model` a `RawModel`? It looks like the `build_ip_adapter` function returns one of a few different `IPAdapter` classes.

Reply:
`build_ip_adapter` returns either an `IPAdapter` or an `IPAdapterPlus`. `IPAdapter` is a subclass of `RawModel`, and `IPAdapterPlus` is a subclass of `IPAdapter`. So it works. Do you want a change to the more evolved class or union?